docs/shared/reference/nx-json.md

```diff
@@ -215,6 +217,30 @@ When defining any options or configurations inside of a target default, you may
 Note that the inputs and outputs are respecified on the @nx/js:tsc default configuration. This is **required**, as when reading target defaults Nx will only ever look at one key. If there is a default configuration based on the executor used, it will be read first. If not, Nx will fall back to looking at the configuration based on target name. For instance, running `nx build project` will read the options from `targetDefaults[@nx/js:tsc]` if the target configuration for build uses the @nx/js:tsc executor. It **would not** read any of the configuration from the `build` target default configuration unless the executor does not match.
 {% /callout %}
 
+#### Cache
+
+In Nx 17 and higher, caching is configured by specifying `"cache": true` in the target default configuration. This will tell Nx that it's ok to cache the results of a given target. For instance, if you have a target that runs tests, you can specify `"cache": true` in the target default configuration for `test` and Nx will cache the results of running tests.
+
+```json {% fileName="nx.json" %}
+{
+  "targetDefaults": {
+    "test": {
+      "cache": true
+    }
+  }
+}
+```
+
+<!-- TODO (@AgentEnder) remove this once we confirm cloud is working with task-level cache property, and update packages/nx/src/tasks-runner/task-orchestrator.ts#isCacheableTask to allow project level cache. -->
+
+{% callout type="warning" title="Caching must be on `targetDefaults`" %}
+
+Nx currently only supports caching on target defaults. If you specify `"cache": true` on a target configuration (e.g. in `project.json`), it will be ignored. This also means that if you specify `"cache": false` in project configuration, it will still be cached if the target default configuration has `"cache": true`.
+
+Additionally, caching must be configured **based on the target name**. While other properties can be read based on the executor for a given task, caching is always read based on the target name. For instance, if you have a target that uses the `@nx/js:tsc` executor, and you have a target default configuration for `build` that specifies `"cache": true`, Nx will cache the results of running that target. However, if you have a target that uses the `@nx/js:tsc` executor, and you have a target default configuration for `@nx/js:tsc` that specifies `"cache": true`, Nx will **not** cache the results of running that target.
```

Comment:
I don't understand this paragraph.

Reply:
Let's chat some tomorrow evening (maybe around 4?) and we can work on rewording it together

Reply:
I understand it now.  I didn't realise that you could create a target default for an executor.  That docs change was added in December of last year, but it only got approval from Victor, not anyone on the docs team.

Reply:
Ah, yeah thats possible. This entire paragraph can go away now though, since we will be able to support cacheability on a per target basis 🎉